src/test_l1_dsHdmiIn.c

```diff
@@ -2564,6 +2565,113 @@ void test_l1_dsHdmiIn_negative_dsGetEdid2AllmSupport_sink(void) {
     UT_LOG("\n Out %s\n", __FUNCTION__);
 }
 
+/**
+ * @brief Ensure dsGetHdmiVersion() correctly retrieves the HDMI compatibility version during positive scenarios.
+ *
+ * **Test Group ID:** Basic: 01@n
+ * **Test Case ID:** 051@n
+ *
+ * **Dependencies:** None@n
+ * **User Interaction:** None
+ *
+ * **Test Procedure:**@n
+ * |Variation / Step|Description|Test Data|Expected Result|Notes|
+ * |:--:|-----------|----------|--------------|-----|
+ * |01|Initialize the HDMI input sub-system using dsHdmiInInit() | | dsERR_NONE | Should Pass |
+ * |02|Call dsGetHdmiVersion() with all valid ports |[Valid Port], dsHdmiMaxCapabilityVersion_t*| dsERR_NONE | Should Pass |
+ * |03|Compare the values with value read from yaml file and make sure they match || Success | The values should be the same |
+ * |04|Call dsHdmiInTerm() to ensure deinitialization | | dsERR_NONE | Clean up after test |
+ *
+ */
+void test_l1_dsHdmiIn_positive_dsGetHdmiVersion(void) {
+
+    gTestID = 51;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+    dsError_t  result = dsERR_NONE;
+    dsHdmiMaxCapabilityVersion_t version = HDMI_COMPATIBILITY_VERSION_MAX;
+    uint8_t numInputPorts = 0;
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+    char keyString[DS_HDMIIN_KEY_SIZE] = {0};
+
+    // Step 1: Initialize the HDMI input sub-system using dsHdmiInInit()
+    UT_ASSERT_EQUAL_FATAL(dsHdmiInInit(), dsERR_NONE);
+
+    if (gSourceType == 0) {
+       // Step 2: Call dsGetHdmiVersion() with valid values (dsHDMI_IN_PORT_0, dsHdmiMaxCapabilityVersion_t*)
+       for (int port = dsHDMI_IN_PORT_0; port < numInputPorts; port++) {
+            result = dsGetHdmiVersion(port, &version);
+            UT_ASSERT_EQUAL(result, dsERR_NONE);
+
+            // Step 3: compare the version with value read from yaml file
+            UT_LOG("version : %d\n", version);
+            snprintf(keyString, DS_HDMIIN_KEY_SIZE, "dsHdmiIn/HdmiCompatibilityVersion/%d", port);
+            UT_ASSERT_KVP_EQUAL_PROFILE_UINT32(version, keyString);
+        }
+    } else if (gSourceType == 1) {
+       // Step 4: Call dsGetHdmiVersion() with valid ports
+       UT_ASSERT_EQUAL(dsGetHdmiVersion(dsHDMI_IN_PORT_0, &version), dsERR_OPERATION_NOT_SUPPORTED);
```

Comment:
you need to check for all the ports here as well.  This may be true if only one port is existing as you had fixed this to port 0

Reply:
updated looping for all ports